packages/codemirror/src/editor.ts

```diff
@@ -871,6 +871,210 @@ class CodeMirrorEditor implements CodeEditor.IEditor {
 export
 namespace CodeMirrorEditor {
   /**
+   * The options used to initialize a code mirror editor.
+   */
+  export
+  interface IOptions extends CodeEditor.IOptions {
+    /**
+     * The mode to use. When not given, this will default to the first mode
+     * that was loaded.
+     */
+    mode?: string | Mode.ISpec;
+
+    /**
+     * The theme to style the editor with.
+     * You must make sure the CSS file defining the corresponding
+     * .cm-s-[name] styles is loaded.
+     * The default is "jupyter".
+     */
+    theme?: string;
+
+    /**
+     * How many spaces a block (whatever that means in the edited language)
+     * should be indented. The default is 2.
+     */
+    indentUnit?: number;
+
+    /**
+     * Whether to use the context-sensitive indentation that the mode provides
+     * (or just indent the same as the line before). Defaults to true.
+     */
+    smartIndent?: boolean;
+
+    /**
+     * The width of a tab character. Defaults to 4.
+     */
+    tabSize?: number;
+
+    /**
+     * Whether, when indenting, the first N*tabSize spaces should be replaced
+     * by N tabs. Default is false.
+     */
+    indentWithTabs?: boolean;
```

Comment:
Monaco has `insertSpaces` and `tabSize` options.  I think we should adopt those as core options and tie `indentUnit` to equal `tabSize`.  I'm hard pressed to think of when they should differ.